pkg/controller/migcluster/migcluster_controller.go

```diff
@@ -155,6 +145,15 @@ func (r *ReconcileMigCluster) Reconcile(request reconcile.Request) (reconcile.Re
 			retry = r.retryFinalizer(cluster)
 		}
 		return reconcile.Result{Requeue: retry}, nil
+	} else {
```

Comment:
According to the [The Kubebuilder Book](https://book.kubebuilder.io/reference/using-finalizers.html) it's a better practice to add the finalizer only when the `DeletionTimestamp` not set.  Makes sense.